edx_exams/apps/api/v1/tests/test_views.py

```diff
@@ -1851,3 +1861,45 @@ def test_get_empty_response(self):
         response = self.request_api('get', self.user, 'course-v1:edx+no+allowances')
         self.assertEqual(response.status_code, 200)
         self.assertEqual(response.data, [])
+
+    def test_post_allowances(self):
+        """
+        Test that the endpoint creates allowances for the given request data
+        """
+        other_exam_in_course = ExamFactory.create(course_id=self.exam.course_id)
+        other_user = UserFactory()
+        StudentAllowanceFactory.create(
+            exam=self.exam,
+            user=self.user,
+            extra_time_mins=30,
+        )
+        StudentAllowanceFactory.create(
+            exam=other_exam_in_course,
+            user=self.user,
+            extra_time_mins=30,
+        )
+
+        request_data = [
+            {'exam_id': self.exam.id, 'user_id': self.user.id, 'extra_time_mins': 45},
+            {'exam_id': other_exam_in_course.id, 'user_id': self.user.id, 'extra_time_mins': 45},
+            {'exam_id': other_exam_in_course.id, 'user_id': other_user.id, 'extra_time_mins': 45},
+        ]
+
+        response = self.request_api('post', self.user, self.exam.course_id, data=request_data)
+        self.assertEqual(response.status_code, 200)
+
+        course_allowances = StudentAllowance.objects.all()
+        self.assertEqual(len(course_allowances), 3)
+
+        self.assertEqual(len(StudentAllowance.objects.filter(user_id=self.user.id)), 2)
+        self.assertEqual(len(StudentAllowance.objects.filter(extra_time_mins=45)), 3)
+
+    def test_invalid_post(self):
+        """
+        Test that 400 response is returned if serializer is invalid
```

Comment:
Can you add a test where none of the expected query strings are provided? No `exam_id`, no `user_id`, and no `extra_time_mins`? What about test that some querystrings are missing?